iaso/odk/validator.py

```diff
@@ -99,7 +99,7 @@ def group_by_lambda(collection, field_name_lambda):
 
 
 def validate_xls_form(xls_file):
-    file_extension = os.path.splitext(xls_file.name)[1].lower()
+    file_extension = os.path.splitext(getattr(xls_file, "name", "form.xlsx"))[1].lower()
```

Comment:
Can you explain this change?

Reply:
The original code xls_file.name assumes that the xls_file object will always have a name attribute. This could fail if:
- The file object doesn't have a name attribute
- The file object is a BytesIO or other in-memory file object
- The file object is a mock or test object
The new code getattr(xls_file, "name", "form.xlsx") is safer because:
- It uses getattr() which safely tries to get the name attribute
- If the name attribute doesn't exist, it falls back to the default value "form.xlsx"
- This prevents potential AttributeError exceptions